src/mender-update/context/context.cpp

```diff
@@ -243,30 +242,24 @@ static error::Error FilterProvides(
 		string escaped;
 		// Potential to escape every character, though unlikely.
 		escaped.reserve(to_clear.size() * 2);
-		// Notable exception: '*', since it has special handling as a glob character.
-		string meta_characters {".^$+()[]{}|?"};
+
+		bool wildcard = false;
 		for (const auto chr : to_clear) {
 			if (chr == '*') {
-				// Turn every '*' glob wildcard into '.*' regex wildcard.
-				escaped.push_back('.');
-			} else if (any_of(meta_characters.begin(), meta_characters.end(), [chr](char c) {
-						   return chr == c;
-					   })) {
-				// Escape every regex special character except '*'.
-				escaped.push_back('\\');
+				// Acts as a regex wildcard, break so `*` isn't included,
+				// this allows us to use the substring to match like a regex
+				wildcard = true;
+				break;
```

Comment:
Lluis and I talked about this, and we think it's ok if we don't support a `*` in the middle of a string, for example `test_*_stuff`. It's difficult to do this kind of matching, and there are few use cases for it. And that is indeed what is done here. However, there are some important details to fix:

1. It needs to test that nothing comes after the wildcard. It's ok to simply return an error for this, explaining that it is not supported (you can use the `errc::not_supported` error code for this, just grep in the code). Make sure there is a test for this below as well.
2. It's a behavior change that you can no longer have a wildcard in the middle of the string, and the changelog should mention this.

@lluiscampos: Although technically this is a breaking change, I think we are ok with treating it as a small change and not triggering a new major release, right? Given its likely very low impact.

Reply:
This was bugging me over the weekend. I find that I am actually a bit uneasy that we are breaking compatibility by only allowing one wildcard at the end. Even if it is an uncommon use case to use anything else. The functionality has been out there for 5+ years already.

But maybe it's not so hard to fix. We really only need to split the string on each wildcard, and search for them one by one. For example in the case of `abc*def*ghi`, we start by searching `abc`, then search for `def` in the rest of the string, then `ghi` in the rest of that string again. There are some extra details such as making sure that the start and end are anchored to the start and end of string, but this should not be hard.